src/shared/util.js

```diff
@@ -1214,24 +1214,50 @@ var createObjectURL = (function createObjectURLClosure() {
   };
 })();
 
+function resolveCall(fn, args, thisArg = null) {
+  if (!fn) {
+    return Promise.resolve(undefined);
+  }
+  return new Promise((resolve, reject) => {
+    resolve(fn.apply(thisArg, args));
+  });
+}
+
+function resolveOrReject(capability, success, reason) {
+  if (success) {
+    capability.resolve();
+  } else {
+    capability.reject(reason);
+  }
+}
+
+function finalize(promise) {
```

Comment:
Unless I'm missing something, you're *not* actually passing `Promise`s to this function (at line https://github.com/mozilla/pdf.js/pull/8430/files#diff-8d2b578d58d4d8c8a2975020fbbc84d9R1492 and below).

Rather, it seems that what you're passing is actually `promiseCapabilities`, which means that this function currently doesn't really make sense to me; what am I missing here?